src/libLocalDescriptorCache/DescDbProxy.cpp

```diff
@@ -410,4 +418,8 @@ DescDbProxy::DescDbProxy(bool disableCache, const std::string &hostAddress, std:
     }
 }
 
-DescDbProxy::~DescDbProxy(){}
\ No newline at end of file
+DescDbProxy::~DescDbProxy()
+{
+    closeDescDbStream(); //close it here to ensure invalid cache file is removed if an error occurs.
```

Comment:
well, isn't destructor called each time gwhisper terminates?

If so, we would invalidate the cache each time. This would make the cache use-less :P It's only purpose is to transport descriptr info from one invocation the the next.
(e.g. every time a user presses <TAB> gwhisper is started afresh and parses the input / gets descriptors without cache)

Reply:
yes the proxy dtor is called evrytime.
actually the reflection stream if already closed gracefully will be make the close at proxy dtor a noop since we check for nullptr.
The cache is only invalidated if we fail to close the stream in time. so we don't disable it and the functional tests are green to prove it.

Reply:
Ok, understand. Maybe add a comment in the destructor.
Then it should be OK